### PR TITLE
Object: implement Annex B: Additional Properties of Object.prototype

### PR DESCRIPTION
**Pull request by ExE-Boss**
This implements [Annex B: Additional Properties of the Object.prototype Object](https://tc39.es/ecma262/#sec-additional-properties-of-the-object.prototype-object) behind the `Object.prototype.__AnnexB__` feature flag.

---

**Refs:** <https://github.com/engine262/engine262/issues/56>

### CODE REVIEW

File: src/engine.mjs [hunk 1]
```diff
@@ -25,6 +25,10 @@ import { CallSite } from './helpers.mjs';
 import * as messages from './messages.mjs';
 
 export const FEATURES = Object.freeze([
+  {
+    name: 'Object.prototype.__AnnexB__',
```
Comment (devsnek) [inline src/engine.mjs:29]: can this just be `'annexB'`

File: src/intrinsics/ObjectPrototype.mjs [hunk 2]
```diff
@@ -94,10 +102,112 @@ function ObjectProto_valueOf(argList, { thisValue }) {
...
+}
+
+// #sec-set-object.prototype.__proto__
+function ObjectProto_protoSetter([proto = Value.undefined], { thisValue }) {
+  // 1. Let O be ? RequireObjectCoercible(this value).
+  const O = Q(RequireObjectCoercible(thisValue));
+
+  // 2. If Type(proto) is neither Object nor Null, return undefined.
+  // 3. If Type(O) is not Object, return undefined.
+  if ((Type(proto) !== 'Object' && Type(proto) !== 'Null') || Type(O) !== 'Object') {
```
Comment (devsnek) [inline src/intrinsics/ObjectPrototype.mjs:118]: please keep these as separate steps

File: src/intrinsics/ObjectPrototype.mjs [hunk 3]
```diff
@@ -94,10 +102,112 @@ function ObjectProto_valueOf(argList, { thisValue }) {
...
+    return surroundingAgent.Throw('TypeError', 'ObjectSetPrototype');
+  }
+
+  // 6. Return undefined.
+  return Value.undefined;
+}
+
+// #sec-object.prototype.__defineGetter__
+function ObjectProto_defineGetter([P, getter], { thisValue }) {
+  const O = Q(ToObject(thisValue));
```
Comment (devsnek) [inline src/intrinsics/ObjectPrototype.mjs:136]: please include the algorithm comments

**Comment by devsnek** [comment]
Can you target the `parser` branch instead of `master`? We will also need to figure out some sort of way to run the tests.

File: src/engine.mjs (hunk 1 above)
Comment (ExE-Boss) [reply src/engine.mjs:29]: I’d rather make it possible to pick and choose which **Annex B** features to include.

**Comment by ExE-Boss** [comment]
@devsnek Moved to: <https://github.com/engine262/engine262/pull/113>

**Pull request closed without merging**